aries-sdk-core/src/main/java/com/hyperledger/aries/messaging/connections/messages/AgentMessage.java

```diff
@@ -0,0 +1,40 @@
+package com.hyperledger.aries.messaging.connections.messages;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+
+public class AgentMessage {
+    @SerializedName("@type")
+    private String type;
+
+    @SerializedName("@id")
+    private String id;
+
+    AgentMessage(String type, String id) {
+        this.type = type;
+        this.id = id;
+    }
+
+    public static AgentMessage from(String rawJson) {
+        try {
+            AgentMessage message = new Gson().fromJson(rawJson, AgentMessage.class);
```

Comment:
Can we follow Java 8 syntax? Use Optional?

Reply:
@sukalpomitra 
- Optional was applied
- The AgentMessage class moved from the com.hyperledger.aries.messaging package.connections.messages to com.hyperledger.aries.messaging package